docs/howto/migrate/migration-guide.md

```diff
@@ -0,0 +1,591 @@
+---
+id: zio-2.x-migration-guide
+title: "ZIO 2.x Migration Guide"
+---
+```scala mdoc:invisible
+import zio._
+```
+
+## Upgrading Dependencies
+
+If we are using one of the following dependencies, we need to upgrade them to the `2.0.0-M2` version:
+
+```scala
+libraryDependencies += "dev.zio" %% "zio"         % "2.0.0-M2"
+libraryDependencies += "dev.zio" %% "zio-streams" % "2.0.0-M2"
+libraryDependencies += "dev.zio" %% "zio-test"    % "2.0.0-M2"
+```
+
+## Automated Scalafix Rules
+TODO
+
+## ZLayer
+
+### Functions to Layers
+
+In ZIO 1.x, when we want to write a service that depends on other services, we need to use `ZLayer.fromSrevice*` variants with a lot of boilerplate:
+
+```scala
+val live: URLayer[Clock with Console, Logging] =
+  ZLayer.fromServices[Clock.Service, Console.Service, Logging.Service] {
+    (clock: Clock.Service, console: Console.Service) =>
+      new Service {
+        override def log(line: String): UIO[Unit] =
+          for {
+            current <- clock.currentDateTime.orDie
+            _ <- console.putStrLn(current.toString + "--" + line).orDie
+          } yield ()
+      }
+  }
+```
+
+ZIO 2.x deprecates all `ZLayer.fromService*` functions:
+
+| ZIO 1.0                        | ZIO 2.x |
+|--------------------------------|---------|
+| ZLayer.fromService             | toLayer |
+| ZLayer.fromServices            | toLayer |
+| ZLayer.fromServiceM            | toLayer |
+| ZLayer.fromServicesM           | toLayer |
+| ZLayer.fromServiceManaged      | toLayer |
+| ZLayer.fromServicesManaged     | toLayer |
+| ZLayer.fromServiceMany         | toLayer |
+| ZLayer.fromServicesMany        | toLayer |
+| ZLayer.fromServiceManyM        | toLayer |
+| ZLayer.fromServicesManyM       | toLayer |
+| ZLayer.fromServiceManyManaged  | toLayer |
+| ZLayer.fromServicesManyManaged | toLayer |
+
+Instead, it provides the `toLayer` extension methods for functions:
+
+```scala
+case class LoggingLive(console: Console, clock: Clock) extends Logging {
+  override def log(line: String): UIO[Unit] =
+    for {
+      current <- clock.currentDateTime.orDie
+      _       <- console.putStrLn(current.toString + "--" + line).orDie
+    } yield ()
+}
+
+object LoggingLive {
+  val layer: URLayer[Has[Console] with Has[Clock], Has[Logging]] =
+    (LoggingLive(_, _)).toLayer
```

Comment:
In practice, I've found it can help to annotate this, such that you can have the IDE [re]generate the type without narrowing it to `LoggingLive`.

```suggestion
    (LoggingLive(_, _)).toLayer[Logging]
```
```